core/quorum.go

```diff
@@ -47,10 +50,35 @@ func NewQuorumManager(s storage.Storage, log logger.Logger) (*QuorumManager, err
 }
 
 // GetQuorum will get the configured or available quorum
-func (qm *QuorumManager) GetQuorum(t int) []string {
+func (qm *QuorumManager) GetQuorum(t int, lastChar string) []string {
+	//QuorumTypeOne is to select quorums from the public pool of quorums instead of a private subnet.
+	//Once a new node is created, it will create a DID. Using the command "registerdid", the peerID and DID will be
+	//published in the network, and all the nodes listening to the subscription will have the DID added on the DIDPeerTable
+	//A new variable quorumList is created, which will contain all the nodes which has DID with same last character as Transaction ID.
+	//It would throw an error if it cannot find any relevant data of if the number of nodes is less than 5.
+	//Then a separate array of type String called quorumAddrList, which will simply contain the address of nodes i.e. PeerID.DID
+	//"quorumAddrList" is returned and checking the availability of nodes would be done in initiateConsensus function in quorum_initiator.go.
 	switch t {
 	case QuorumTypeOne:
-		return nil
+		var quorumList []wallet.DIDPeerMap
+		err := qm.s.Read(wallet.DIDPeerStorage, &quorumList, "did_char=?", lastChar)
+		fmt.Println(quorumList)
+
+		if err != nil {
+			qm.log.Error("Quorums not present")
+			return nil
+		}
+		if len(quorumList) < 5 {
+			qm.log.Error("Not enough quorums present")
+			return nil
+		}
+		var quorumAddrList []string
+		fmt.Println(quorumAddrList)
+		for _, q := range quorumList {
```

Comment:
If the `quorumAddrList` returned is greater than 7 , Wouldn't the sender connect to all available quorum instead of upper limit 7? Either the append operation can be stopped once `quorumAddrList` length reaches 7 or handled at `initiateConsensus()` by connecting to quorum and checking for balance.

Reply:
Yes, I agree. The sender will try to connect to all the available quorums.
If we stop the append operation once the length is reached, we might have quorums that are offline. In this case, we'll have to get the list again.
After we have the list of quorums, we are trying to connect to the quorums and pledge tokens. But yes, a check for balance would be important. In initiateConsensus(), we can put a check probably where we are trying to connect with quorums, I would appreciate if we can discuss this once before making any changes. 
Another thing related to quorum count, from what I understand, there are no checks for type2 quorums as well, do we need to check that as well?

Reply:
Is this resolved ? @ashi31  . Right now the sender will try to connect to all nodes satisfying the condition. 
Ideally You could do a generic ping check to make sure nodes are alive( This has to be done for both type 1 and type2) 
or as temporary fix only pick 7.

Reply:
Hi, The temporary fix to select 7 quorums is added. And the datatoken file is reverted back to original.